spec/requests/vendor_api/get_multiple_applications_spec.rb

```diff
@@ -2,26 +2,37 @@
 
 RSpec.describe 'Vendor API - GET /api/v1/applications', type: :request do
   include VendorApiSpecHelpers
+  include FactoryHelpers
 
-  it 'returns applications of the authenticating provider' do
+  it 'returns applications scoped by `provider`' do
     create_list(
-      :application_choice, 2,
-      course: create(:course, provider: currently_authenticated_provider)
+      :application_choice,
+      2,
+      course_option: course_option_for_provider(provider: currently_authenticated_provider),
     )
 
-    create(:application_choice)
-
-    get_api_request "/api/v1/applications?since=#{(Time.now - 1.days).iso8601}"
+    create_list(
+      :application_choice,
+      1,
+      course_option: course_option_for_provider(provider: alternate_provider),
+    )
 
+    get_api_request "/api/v1/applications?provider_ucas_code=UCL&since=#{(Time.now - 1.days).iso8601}"
     expect(parsed_response['data'].size).to be(2)
   end
 
   it 'returns applications filtered with `since`' do
     Timecop.travel(Time.now - 2.days) do
-      create(:application_choice, course: create(:course, provider: currently_authenticated_provider))
+      create(
```

Comment:
There is a lot of duplication so yes. I'm not sure we can use the same memoised pattern as `currently_authenticated_provider` because some specs set up more than one application choice. Does it make sense to just create a helper called `create_application_choice_for_currently_authenticated_provider`?

Reply:
Yes, perfect!